pandas/io/sql.py

```diff
@@ -2066,8 +2114,8 @@ def read_query(
         sql,
         index_col=None,
         coerce_float: bool = True,
-        params=None,
```

Comment:
Why did the ordering of `params` change?

Reply:
When I changed the return type of `pandasSQL_builder` to `Iterator[PandasSQL]`, I ran into some code check errors from mypy.  I also tried `Iterator[SQLDatabase] | Iterator[SQLiteDatabase]`.  The way I got mypy checks to pass was by making everything consistent, including argument order, between `SQLDatabase` and `SQLiteDatabase`.

Reply:
@mroeschke Would it help if I split the following commit out of this PR?  https://github.com/cdcadman/pandas/commit/32fa4cfe23be497ba56af9239cd9f65e03537500

Reply:
> @mroeschke Would it help if I split the following commit out of this PR? [cdcadman@32fa4cf](https://github.com/cdcadman/pandas/commit/32fa4cfe23be497ba56af9239cd9f65e03537500)

Yes definitely!

Reply:
I split out this change in https://github.com/pandas-dev/pandas/pull/49587 .